code/controllers/subsystem/squeak.dm

```diff
@@ -35,6 +37,10 @@ var/datum/subsystem/squeak/SSsqueak
 	exposed_wires.Cut()
 
 	var/list/all_turfs = block(locate(1,1,1), locate(world.maxx,world.maxy,1))
-	for(var/turf/open/floor/plating/T in all_turfs)
-		if(locate(/obj/structure/cable) in T)
-			exposed_wires += T
+	turf_loop:
+		for(var/turf/open/floor/plating/T in all_turfs)
+			if(locate(/obj/structure/cable) in T)
+				for(var/atom/A in T)
+					if(A.density)
```

Comment:
there's a proc for this!! `is_blocked_turf(turf_in_question)`, returns TRUE if turf is blocked